internal/sapcontrol/current_instance.go

```diff
@@ -0,0 +1,58 @@
+package sapcontrol
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// this structure will be used for static labels, common to all the metrics
+type CurrentSapInstance struct {
+	SID      string
+	Number   int32
+	Name     string
+	Hostname string
+}
+
+func (i *CurrentSapInstance) String() string {
+	return fmt.Sprintf("SID: %s, Name: %s, Number: %d, Hostname: %s", i.SID, i.Name, i.Number, i.Hostname)
+}
+
+func (s *webService) GetCurrentInstance() (*CurrentSapInstance, error) {
+	var err error
+
+	// since the information we want here doesn't change over time, we only want to execute the remote call once
+	s.once.Do(func() {
+		var response *GetInstancePropertiesResponse
+		response, err = s.GetInstanceProperties()
+
+		if err != nil {
+			err = errors.Wrap(err, "could not performGetInstanceProperties query")
+			return
+		}
+
+		s.currentSapInstance = &CurrentSapInstance{}
+
+		for _, prop := range response.Properties {
+			switch prop.Property {
+			case "SAPSYSTEM":
+				var num int
```

Comment:
do we need this declaration?

Reply:
yes, because we need to reuse `err` from outside the anonymous function, so we can't use `num, err := etc` because it would declare a new `err`, masking the outer one.